src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -693,7 +704,7 @@ protected virtual JwtSecurityToken ValidateSignature(string token, TokenValidati
                 {
                     try
                     {
-                        if (ValidateSignature(encodedBytes, signatureBytes, securityKey, jwt.Header.Alg))
+                        if (ValidateSignature(encodedBytes, signatureBytes, securityKey, algorithm))
```

Comment:
I think what @brentschmaltz is saying is that we are not consistent with the value of algorithm. It should be jwt.header.alg instead of signingCredentials.algorithm